### DISCUSSION TITLE
Support alpine-typescript component names

### DISCUSSION THREAD

**Discussion by stancl** [alpine-collective/alpinejs-devtools#173] [Remnant]
I have some components that look like this:

```html
<a href="..." x-data="Alpine.component('modalLink')('index', 'products')" x-spread="click">
    Products
</a>
```

alpinejs-devtools renders it like this:

<img width="595" alt="Screen Shot 2021-03-09 at 14 08 13" src="https://user-images.githubusercontent.com/33033094/110475220-e3ee3180-80e0-11eb-9765-b884ed87a106.png">

Also this:

```js
Alpine.component('darkModeToggle')()
```

is rendered as `Alpine.component`

Is there some behavior for only looking at the last `()`? I think I can solve this on my end but maybe this could be an issue for some other components too. Wondering what's the rationale behind the different display name here.

Thanks

**Comment by HugoDF** [comment]
Feel free to PR

The function that gets the function name from `x-data` value is at https://www.github.com/alpine-collective/alpinejs-devtools/tree/master/packages%2Fshell-chrome%2Fsrc%2Futils.js#L269

**Reply by stancl** [reply]
Ah, I see, so it's meant to make function calls look nicer? So it removes the parameters, which transforms `myFunction('foo')` to `myFunction`?

**Reply by ryangjchandler** [reply]
@stancl Yeah, that's it.

**Reply by HugoDF** [reply]
yeah we might want to skip the formatting if it's `something.fn()`, just print the whole thing I guess

or update the regex to remove all `(*)` except the first one if there's more than 1 (all if there's only 1 instance)?

**Reply by stancl** [reply]
I added an answer with userland code that adds nice support here. I agree that if there's something more complex than `foo(bar)` it could be nice to show the full thing. But as far as components made with my package go, I think the solution below is probably ideal. Showing full data can be problematic and this mimics the behavior of e.g. Vue components in Vue devtools.

**Comment by stancl** [comment]
This seems like the best solution:

```js
import { AlpineComponent } from '@leanadmin/alpine-typescript';

window.Alpine.onBeforeComponentInitialized(component => {
    if (! component.$el.hasAttribute('x-title')) {
        if (component.$data.constructor.prototype instanceof AlpineComponent) {
            component.$el.setAttribute('x-title', component.$data.constructor.name);
        }
    }
});
```

<img width="276" alt="Screen Shot 2021-03-09 at 19 23 52" src="https://user-images.githubusercontent.com/33033094/110518895-fed69b00-810c-11eb-8b54-456ee69f125f.png">

**Reply by HugoDF** [reply]
nice, so solved in `alpine-typescript` 👍
